### DISCUSSION TITLE
react-three/drei@9.77.7 - "TypeError: a is not a function"

### DISCUSSION THREAD

**Discussion by JHansen2000** [pmndrs/drei#1546] [General]
I just installed the latest release of react-three/drei (9.77.7) and am receiving this error when rendering a canvas.
```
- error node_modules/@react-three/drei/index.cjs.js (1:8648) @ eval
- error TypeError: a is not a function
```

```tsx
// threejs/page
import { NextPage } from "next"
import { Canvas } from "@react-three/fiber"
import { OrbitControls } from "@react-three/drei"

const ThreeJSPage : NextPage = () => {
    return (
        <Canvas
            shadows
            camera={{
                position: [4, 3, 0],
            }}
        >
            <ambientLight intensity={0.1} />
            <spotLight position={[3, 5, 2]}/>
            <OrbitControls />
        </Canvas>
    )
}
export default ThreeJSPage
```

This error is resolved when downgrading react-three/drei.

**Comment by jglee96** [comment]
I've been spending time since this afternoon with the same error.
In my case, it appears when I run test (`npx vitest`, `jest`).
Therefore, I make some codesandbox. ([vitest-version](https://codesandbox.io/p/github/jglee96/drei-vitest/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cljhbcupx000b356omelcu87i%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cljhbcupx000d356ohoimfqbf%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cljhbcupx000b356omelcu87i%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cljhbcupx000a356oy82n7g8n%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.gitignore%2522%257D%255D%252C%2522id%2522%253A%2522cljhbcupx000b356omelcu87i%2522%252C%2522activeTabId%2522%253A%2522cljhbcupx000a356oy82n7g8n%2522%257D%252C%2522cljhbcupx000d356ohoimfqbf%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cljhbcupx000c356ofayvy7v7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%2522%257D%255D%252C%2522id%2522%253A%2522cljhbcupx000d356ohoimfqbf%2522%252C%2522activeTabId%2522%253A%2522cljhbcupx000c356ofayvy7v7%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D),[ jest-version](https://codesandbox.io/p/github/jglee96/drei-jest/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cljhb8cdl000b356odu011674%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cljhb8cdl000d356oujk4rsf3%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cljhb8cdl000b356odu011674%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cljhb8cdk000a356o4rvktegh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fvite.config.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cljhb8cdl000b356odu011674%2522%252C%2522activeTabId%2522%253A%2522cljhb8cdk000a356o4rvktegh%2522%257D%252C%2522cljhb8cdl000d356oujk4rsf3%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cljhb8cdl000c356o4y45tb5v%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cljhb8cdl000d356oujk4rsf3%2522%252C%2522activeTabId%2522%253A%2522cljhb8cdl000c356o4y45tb5v%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D))

Also, I found that the "a" is `create` of zustand which is in [here](https://github.com/pmndrs/drei/blob/290272d55a08a8b0fa0c594d0ccb4ba24b06abb7/src/core/useProgress.tsx#L14)


What version did you downgrade to?

**Reply by JHansen2000** [reply]
I downgraded to version [9.76.0](https://github.com/pmndrs/drei/releases/tag/v9.76.0) to fix the issue. Please let me know if you find another way around this problem, though. I would like to keep my packages current wherever possible.

**Comment by jglee96** [comment]
under `v9.77.7` (`v9.77.6` ~)
![image](https://github.com/pmndrs/drei/assets/22130454/68f5fa47-bd3e-487e-bc9b-d88e052b124f)

`v9.77.7`
![image](https://github.com/pmndrs/drei/assets/22130454/cc9e2e0b-194d-4d96-b973-f48ffd86ed09)

The[ v9.77.7](https://github.com/pmndrs/drei/releases/tag/v9.77.7) was released yesterday.
There are rollup changes in the release, and this seems to be causing this issue.

**Reply by jglee96** [reply]
#1544

**Reply by belgattitude** [reply]
9.77.6 works fine. Reproduction with nextjs available in https://github.com/belgattitude/artist/pull/6 (updated to 9.77.7)
